google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1beta2/StreamWriterV2.java

```diff
@@ -0,0 +1,367 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigquery.storage.v1beta2;
+
+import com.google.api.core.ApiFuture;
+import com.google.api.core.SettableApiFuture;
+import com.google.cloud.bigquery.storage.v1beta2.StreamConnection.DoneCallback;
+import com.google.cloud.bigquery.storage.v1beta2.StreamConnection.RequestCallback;
+import com.google.common.base.Preconditions;
+import com.google.common.util.concurrent.Uninterruptibles;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import java.time.Duration;
+import java.util.Deque;
+import java.util.LinkedList;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.logging.Logger;
+import javax.annotation.concurrent.GuardedBy;
+
+/**
+ * A BigQuery Stream Writer that can be used to write data into BigQuery Table.
+ *
+ * <p>TODO: Add credential support.
+ *
+ * <p>TODO: Attach schema.
+ *
+ * <p>TODO: Add max size check.
+ *
+ * <p>TODO: Add inflight control.
+ *
+ * <p>TODO: Attach traceId.
+ *
+ * <p>TODO: Support batching.
+ *
+ * <p>TODO: Support schema change.
```

Comment:
There is another popular request, which is, the user just call append without looking at the future, and there is a FlushAll to flush the inflight queue and throw if there is any error happened. But the priority of it is behind support schema change.

Reply:
I don't think we want to support this at all. It seems very weird. We can talk about this offline.